SD-1511 System should emit `variant.out_of_stock` if Variant is sold out

aleksandarpetrushev opened the pull request:
https://spark-solutions.atlassian.net/browse/SD-1511

Commit:
Emit variant.out_of_stock webhook event

Commit:
Remove unnecessary time freezing in spec

damianlegawiec commented on api/app/models/spree/api/webhooks/stock_movement_decorator.rb:
```diff
@@ -0,0 +1,15 @@
+        def update_stock_item_quantity
+          super
+          variant = stock_item.variant
+          variant.queue_webhooks_requests!('variant.out_of_stock') unless variant.in_stock?
```
@aleksandarpetrushev similar situation as described here: https://github.com/spree/spree/pull/11460#discussion_r742771950

Commit:
Check if variant was in stock before update

The variant.out_of_stock event should be emitted only when the variant
was in stock before the update and now it is not. This is important in
case the variant is out of stock and after an update it still persists 0
as stock. In this case the event was already emitted and should not be
emitted again.

Commit:
Do not use #reload in decorator

damianlegawiec commented on api/spec/models/spree/api/webhooks/stock_movement_decorator_spec.rb:
```diff
@@ -0,0 +1,28 @@
+
+      it 'emits the variant.out_of_stock event' do
+        expect { subject }.to emit_webhook_event('variant.out_of_stock')
+      end
```
@aleksandarpetrushev let's also add a test when count_on_hand is 0 and there is an update which still persists it to 0

Commit:
Check if variant in stock before update

sebastian-palma commented on api/app/models/spree/api/webhooks/stock_movement_decorator.rb:
```diff
@@ -0,0 +1,16 @@
+    module Webhooks
+      module StockMovementDecorator
+        def update_stock_item_quantity
+          variant_in_stock_before_update = stock_item.variant.in_stock?
```
You could make this a private predicate method.

Commit:
Add spec for variant out of stock before and after update
